augur/filter/subsample.py

```diff
@@ -142,7 +142,7 @@ def get_groups_for_subsampling(strains, metadata, group_by=None):
 
             # Generate columns.
             if constants.DATE_YEAR_COLUMN in generated_columns_requested:
-                metadata[constants.DATE_YEAR_COLUMN] = metadata[f'{temp_prefix}year']
+                metadata[constants.DATE_YEAR_COLUMN] = metadata[f'{temp_prefix}year'].astype(str)
```

Comment:
I believe this currently sets the dtype to `object`.  Not sure it if matters, but probably safer to use the pandas.StringDtype here to match the other string columns. 

```suggestion
                metadata[constants.DATE_YEAR_COLUMN] = metadata[f'{temp_prefix}year'].astype('string')
```

Reply:
Based on the [pandas dtypes documentation](https://pandas.pydata.org/docs/user_guide/basics.html#dtypes), `'string'` is an alias for `str` so it shouldn't matter.

<img width="679" alt="image" src="https://github.com/user-attachments/assets/8683277d-ea8a-42ea-a545-3a033c79db8a" />

I'll still change it to `'string'` since that's slightly more readable.

Reply:
Done:

https://github.com/nextstrain/augur/blob/232038e76a362de7ebfaa2885ab2a382d3415201/augur/filter/subsample.py#L145

Reply:
Oh you're right. This snippet from a [doc](https://pandas.pydata.org/pdeps/0014-string-dtype.html) detailing upcoming string dtype changes in pandas 3.0 is the best explanation I found (emphasis mine):

> For the generic string alias to specify the dtype, `"string"` is already used for the `StringDtype` using `pd.NA`. This PDEP proposes to use `"str"` for the new default `StringDtype` using `NaN`. This ensures backwards compatibility for code using `dtype="string"`, and was also chosen because `dtype="str"` or `dtype=str` currently already works to ensure your data is converted to strings (**only using object dtype for the result**).

Reply:
I'll make a separate PR to use `"string"` in the rest of our code.

Reply:
#1782